the-collab-lab/tcl-33-smart-shopping-list
# af jt firebase - Reading and Writing to Firestore!

## Description

Updated dependencies and import statements so that Firebase v9 works with react-firebase-hooks package with no compatibility error. Added a new AddItem component which gets and displays 'lists' collection from Firestore and allows user to add a new item from the inputs entered.

## Related Issue
closes #1 

## Acceptance Criteria

- The following have been added as project dependencies: firebase & react-firebase-hooks
- Make a change in the Firestore database and it shows up in the app
- Make a change in the app and it shows up in the Firestore database

## Type of Changes



|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|    | :hammer: Refactoring       |
|    | :100: Add tests            |
|    | :link: Update dependencies |
|    | :scroll: Docs              |

## Before

<img width="698" alt="Before" src="https://user-images.githubusercontent.com/72953646/136370602-1b776cea-0998-4ca4-8b7b-212265d39545.png">

## After

<img width="972" alt="After" src="https://user-images.githubusercontent.com/72953646/136370616-35a1cb22-ed87-47e2-bb0d-260ad07da111.png">

## Testing Steps / QA Criteria

[Firestore](https://console.firebase.google.com/project/tcl-33-smart-shopping-list/firestore/data/~2Flists~2Foi88oC57FK3owFySt6i4)

1.  Run `npm i` then start the server using `npm start` 
2. Add an item using the form below and check if it was added to Firestore
3. Add an item through the Firestore webpage and check if it appears in the collection list.
